### ISSUE TITLE
Read/write direction mismatch between configuration file and code goes unnoticed

### ISSUE THREAD

**Issue by ajaust** [precice/precice#428]
Hi,

### Brief summary

The read/write direction in the configuration file and code did not match (the directions were interchanged). This lead to the a situation where the second solver would read data that was zero instead of the data that was written by the first solver.

### Question/suggestion

Would it be possible to show a warning, throw an exception or similar if a user has different read/write directions for data in the configuration file and the code?

Due to the configuration file preCICE should know about the data read/write direction (led it be correct or wrong as in my case) and thus should be able to detect a mismatch.

### More detailed summary

I observed the following problem in on of my applications. In my config file I set the wrong directions for the exchange of quantities (`Velocity` and `Pressure` in my case) for the participant `FreeFlow`. It looked like that
```xml
<participant name="FreeFlow">
  ...
  <read-data name="Velocity" mesh="FreeFlowMesh"/>
  <write-data name="Pressure" mesh="FreeFlowMesh"/>
  ...
</participant>
```
In fact, `FreeFlow` should **write** `Velocity` and **read** `Pressure`. This is also done correctly in my code where I have something
```c++
  precice.readBlockScalarData( pressureId, vertexIDs_.size(), vertexIDs_.data(), pressure.data() );
  ...
  precice.writeBlockScalarData( velocityId, vertexIDs_.size(), vertexIDs_.data(), velocity.data() );
```
In the end I had the directions for reading writing in the opposite order as intended. Therefore the other solver would read data that was zero instead of the velocity the solver `FreeFlow` had actually written. 

### Further information
- preCICE 1.4.0 and C++ API

**Comment by fsimonis** [comment]
We noticed and fixed this vulnerability in version 1.5.0 for both Data and Meshes. So, upgrading will solve this problem for you.

From the 1.5.0 changelog:
> Added sanitization of IDs to the SolverInterface and harden it against misconfiguration using a consistent mechanism to express requirements.

**Comment by ajaust** [comment]
Thanks for the quick reply! I guess I have to update preCICE. :)
